### PR TITLE
Add advanced usage documentation and bump version

### PR DESCRIPTION
**Pull request by yadutaf**
Add the documentation related to #43 and ``need_auth`` param which was not documented and prepare to release.

### CODE REVIEW

**Comment by yadutaf** [comment]
cc @geoffreybauduin

File: debian/changelog [hunk 1]
```diff
@@ -16,7 +24,7 @@ python-ovh (0.4.4) trusty; urgency=medium
   * Add debian folder
   * fix: remove Python SNI warnings, OVH API does not need SNI (#35)
   * fix: coveralls version
-  * [auto] bump version to v0.4.6
+  * [auto] bump version to v0.4.7
```
Comment (Rbeuque74) [inline debian/changelog]: ?

Comment (yadutaf) [inline debian/changelog]: Argh, I completely forgot about the packaging... I should probably remove it altogether. Fixing

Comment (yadutaf) [inline debian/changelog]: The debian part of the release script was broken after another PR. Fixed.

**Comment by Rbeuque74** [comment]
ok for merge

**Pull request merged**